nifi-nar-bundles/nifi-prometheus-bundle/nifi-prometheus-reporting-task/src/test/java/org/apache/nifi/reporting/prometheus/TestPrometheusMetricsUtil.java

```diff
@@ -0,0 +1,404 @@
+    public void testAggregatePercentUsedWithNestedProcessGroupsWithCountBackpressure() {
+        final Map<String, Double> aggregatedMetrics = new HashMap<>();
+
+        PrometheusMetricsUtil.aggregatePercentUsed(nestedProcessGroupStatusWithCountBackpressure, aggregatedMetrics);
+
+        assertThat(aggregatedMetrics.size(), equalTo(4));
+        assertThat(EXPECTED_NESTED_BYTES_PERCENT_VALUE, equalTo(aggregatedMetrics.get(NIFI_PERCENT_USED_BYTES)));
+        assertThat(EXPECTED_BACKPRESSURE_PERCENT_USED_VALUE, equalTo(aggregatedMetrics.get(NIFI_PERCENT_USED_COUNT)));
+        assertThat(EXPECTED_FALSE_BACKPRESSURE, equalTo(aggregatedMetrics.get(BYTES_AT_BACKPRESSURE)));
+        assertThat(EXPECTED_TRUE_BACKPRESSURE, equalTo(aggregatedMetrics.get(COUNT_AT_BACKPRESSURE)));
+    }
+
+    @Test
+    public void testAggregateConnectionPredictionsWithMixedValues() {
+        Map<String, Double> aggregatedMetrics = new HashMap<>();
+        generateConnectionAnalyticMetricsAggregation(aggregatedMetrics, mixedValuedPredictions);
+
+        assertThat(aggregatedMetrics.size(), equalTo(2));
+        assertThat(1.0, equalTo(aggregatedMetrics.get(NIFI_TIME_TO_BYTES_BACKPRESSURE_PREDICTION)));
+        assertThat(2.0, equalTo(aggregatedMetrics.get(NIFI_TIME_TO_COUNT_BACKPRESSURE_PREDICTION)));
+    }
+
+    @Test
+    public void testAggregateConnectionPredictionsWithAllDefaultValues() {
+        Map<String, Double> aggregatedMetrics = new HashMap<>();
+        generateConnectionAnalyticMetricsAggregation(aggregatedMetrics, defaultValuedPredictions);
+
+        assertThat(aggregatedMetrics.size(), equalTo(2));
+        assertThat(EXPECTED_DEFAULT_PREDICTION_VALUE, equalTo(aggregatedMetrics.get(NIFI_TIME_TO_BYTES_BACKPRESSURE_PREDICTION)));
+        assertThat(EXPECTED_DEFAULT_PREDICTION_VALUE, equalTo(aggregatedMetrics.get(NIFI_TIME_TO_COUNT_BACKPRESSURE_PREDICTION)));
+    }
+
+    @Test
+    public void testAggregateConnectionPredictionsWithBackpressure() {
+        Map<String, Double> aggregatedMetrics = new HashMap<>();
+        aggregatedMetrics.put(BYTES_AT_BACKPRESSURE, 1.0);
+        aggregatedMetrics.put(COUNT_AT_BACKPRESSURE, 0.0);
+        generateConnectionAnalyticMetricsAggregation(aggregatedMetrics, mixedValuedPredictions);
+
+        assertThat(EXPECTED_BACKPRESSURE_PREDICTION_VALUE, equalTo(aggregatedMetrics.get(NIFI_TIME_TO_BYTES_BACKPRESSURE_PREDICTION)));
+        assertThat(2.0, equalTo(aggregatedMetrics.get(NIFI_TIME_TO_COUNT_BACKPRESSURE_PREDICTION)));
+    }
+
+    @Test
+    public void testAggregatedConnectionPredictionsDatapointCreationWithAnalyticsNotSet() {
+        ConnectionAnalyticsMetricsRegistry connectionAnalyticsMetricsRegistry = new ConnectionAnalyticsMetricsRegistry();
+        Map<String, Double> emptyAggregatedMetrics = new HashMap<>();
+
+        PrometheusMetricsUtil.createAggregatedConnectionStatusAnalyticsMetrics(connectionAnalyticsMetricsRegistry,
+                emptyAggregatedMetrics,
+                "",
```

Comment:
Recommend defining a static constant named `EMPTY`, or using a static import of `StringUtils.EMPTY` as opposed to having the empty string quotes repeated through multiple methods.